File: mns-rest/src/test/java/cz/fb/manaus/rest/TestTask.java
package cz.fb.manaus.rest;

import cz.fb.manaus.core.maintanance.ConfigUpdate;
import cz.fb.manaus.core.maintanance.PeriodicMaintenanceTask;
import org.springframework.stereotype.Component;

import java.time.Duration;

@Component
class TestTask implements PeriodicMaintenanceTask {

    @Override
    public String getName() {
        return "testTask";
    }

    @Override
    public Duration getPausePeriod() {
        return Duration.ofNanos(777);
    }

    @Override
    public ConfigUpdate execute() {
        var command = ConfigUpdate.empty(Duration.ofHours(12));
        command.getDeletePrefixes().add("test_delete");
        return command;
    }
}

File: mns-rest/src/test/java/cz/fb/manaus/rest/TestTask.kt
package cz.fb.manaus.rest

import cz.fb.manaus.core.maintanance.PeriodicTask
import org.springframework.stereotype.Component
import java.time.Duration

@Component
internal class TestTask : PeriodicTask {

    override val name: String = "testTask"

    override val pausePeriod: Duration = Duration.ofNanos(777)

    override fun execute() {

    }
}
